models/intermediate/int_xero__journal_line_pivoted_tracking_categories.sql

```diff
@@ -0,0 +1,61 @@
+{{ config(enabled=(var('xero__using_journal_line_has_tracking_category', True)
+        and var('xero__using_tracking_categories', True))) }}
+
+with journal_line_has_tracking as (
+
+    select *
+    from {{ var('journal_line_has_tracking_category') }}
+),
+
+tracking_category as (
+
+    select *
+    from {{ var('tracking_category') }}
+    where lower(status) in ('active', 'archived')
+),
+
+tracking_category_option as (
+
+    select *
+    from {{ var('tracking_category_option') }}
+    where lower(status) = 'active'
+),
+
+journal_tracking as (
+
+    select
+        journal_line_has_tracking.journal_id,
+        journal_line_has_tracking.journal_line_id,
+        journal_line_has_tracking.source_relation,
+        tracking_category.name as tracking_category_name,
+        tracking_category_option.name as tracking_option_name
+    from journal_line_has_tracking
+
+    left join tracking_category
+        on journal_line_has_tracking.tracking_category_id = tracking_category.tracking_category_id
+        and journal_line_has_tracking.source_relation = tracking_category.source_relation
+
+    left join tracking_category_option
+        on journal_line_has_tracking.tracking_category_option_id = tracking_category_option.tracking_option_id
+        and journal_line_has_tracking.source_relation = tracking_category_option.source_relation
```

Comment:
If this same join (or even an extension of it) is needed in both the journal and invoice intermediate tables then we should consider making the combined result be an ephemeral intermediate model that can then be reused across models.

Reply:
Ah, in case we decide to bring other `*_has_tracking_category` models in. That makes sense. 

Updated.